desktop/main.js

```diff
@@ -147,6 +150,27 @@ const mainWindow = (() => {
                 titleBarStyle: 'hidden',
             });
 
+            if (!isDev) {
+                const newDotURL = isProduction ? 'https://new.expensify.com' : 'https://staging.new.expensify.com';
+
+                // Modify the request origin for requests sent to our API
+                const validDestinationFilters = {urls: ['https://*.expensify.com/*']};
+                browserWindow.webContents.session.webRequest.onBeforeSendHeaders(validDestinationFilters, (details, callback) => {
+                    // eslint-disable-next-line no-param-reassign
```

Comment:
I think you could get around this by using `const newDetails = {...details}`?

Reply:
I actually took one out of your playbook from [this PR](https://github.com/Expensify/App/pull/7649) ... this code will be executed before and after every api request, so performance matters. Using the spread operator would cause us to perform an unnecessary shallow copy of all the headers for every request twice per request, which seems less good than just overwriting a couple properties.

Reply:
Yeah, that's probably a good call. I had also thought of that, and the fact that this is done on every request is a good reason to keep it very performant. The only caveat I will say is that in my PR, I think the spread operator is not performant because of how Babel chooses to transpile it into `_spreadOperator()`. Since this is in Node.js and nothing is transpiled, I would think it's much more performant.

Something else to consider. In my PR, since there was a reducer used along with the spread operator, it could have been compounding the problem. This is mentioned specifically in this SO: https://stackoverflow.com/questions/55843097/does-spread-operator-affect-performance

> beware of using spread with an array.reduce(). I suspect it leads to O(n^2) behavior or worse.

So, I'm not really sure that using a spread operator here is such a bad thing.

Reply:
Thanks, I wasn't aware of that context, especially regarding Babel.

> So, I'm not really sure that using a spread operator here is such a bad thing.

That makes sense – I agree it might not create a performance problem to use the spread operator here. But still I would lean towards leaving this as-is because I think it will be "technically" more performant in a very performance-critical place, which seems worth the couple `eslint-disable`s to me. I don't feel as strongly anymore though, so I'll change it if you prefer.